source/server/active_stream_listener_base.h

```diff
@@ -84,8 +84,13 @@ class ActiveStreamListenerBase : public ActiveListenerImplBase,
 
   void onSocketAccepted(std::unique_ptr<ActiveTcpSocket> active_socket) {
     // Create and run the filters
-    config_->filterChainFactory().createListenerFilterChain(*active_socket);
-    active_socket->continueFilterChain(true);
+    if (config_->filterChainFactory().createListenerFilterChain(*active_socket)) {
+      active_socket->continueFilterChain(true);
+    } else {
+      // If create listener filter chain failed, it means the listener is missing
+      // config due to the ECDS. Then close the connection directly.
+      active_socket->socket_->close();
```

Comment:
Is there a test that validates what happens when the connection is closed, but `active_socket->iter_` isn't at the end of the filters?
Will those filters work as expected if the actual socket is closed?

Reply:
This is going to line 101, which is the same code path when the listener filter rejected the connection. So it should be tested well.

Here is ActiveTcpSocket knew that the listener filter rejected the connection
https://github.com/envoyproxy/envoy/blob/c0bf266187924c780d5cf2d2ffd08dd44f755b5a/source/server/active_tcp_socket.cc#L126-L129

Then it sets the `active_socket->iter_` to the `active_socket->accept_filters_.end()`
https://github.com/envoyproxy/envoy/blob/c0bf266187924c780d5cf2d2ffd08dd44f755b5a/source/server/active_tcp_socket.cc#L159-L164

The initial value of `active_socket->iter_` is `active_socket->accept_filters_.end()`
https://github.com/envoyproxy/envoy/blob/c0bf266187924c780d5cf2d2ffd08dd44f755b5a/source/server/active_tcp_socket.cc#L16

So we close the socket and have `active_socket->iter_` as the end, then it is going to the same code path.

We have integration test cover this 
https://github.com/envoyproxy/envoy/blob/c0bf266187924c780d5cf2d2ffd08dd44f755b5a/test/integration/listener_lds_integration_test.cc#L917

Reply:
I guess that essentially after the call for `close()` one can add an `ASSERT(active_socket->iter_ == active_socket->accept_filters_.end()`.
What confused me was the structure of this function.

Reply:
got it, that is good idea, thanks!

Reply:
LGTM for the changes outside of this function.  For the changes inside this function, I am a little bit not clear what the code does from line 97 to the end and how this change impact them.